core/templates/services/playthrough-issues-backend-api.service.ts

```
@@ -15,87 +15,106 @@
+  // error "TS2345: Argument of type '1' is not assignable to parameter
+  // of type 'string'".
+  fetchIssues(explorationId: string, explorationVersion: any): Promise<string> {
+    if (this.cachedIssues !== null) {
+      return new Promise((resolve) => resolve(this.cachedIssues));
+    } else {
+      return this.httpClient.get(
+        this.getFullIssuesUrl(explorationId), {
+          params: { exp_version: explorationVersion }, observe: 'response'
+        }).toPromise().then((response: any) => {
+        var unresolvedIssueBackendDicts = response.body;
+        this.cachedIssues = unresolvedIssueBackendDicts.map(
+          this.playthroughIssueObjectFactory.createFromBackendDict);
+        return this.cachedIssues;
+      });
+    }
+  }
+
+  fetchPlaythrough(
+      expId: string, playthroughId: string): Promise<PlaythroughIssue> {
+    return this.httpClient.get(
+      this.getFullPlaythroughUrl(expId, playthroughId), { observe: 'response' })
+      .toPromise().then((response: any) => {
+        var playthroughBackendDict = response.body;
+        return this.playthroughIssueObjectFactory.createFromBackendDict(
+          playthroughBackendDict);
+      });
+  }
+
+  resolveIssue(
+      issueToResolve: PlaythroughIssue, expId: string, expVersion: number)
+      : Promise<object> {
+    return this.httpClient.post(
+      this.getFullResolveIssueUrl(expId), {
+        exp_issue_dict: issueToResolve.toBackendDict(),
+        exp_version: expVersion
+      }).toPromise().then(() => {
+      var issueIndex = this.cachedIssues !== null ?
+        this.cachedIssues.findIndex((issue) => {
+          return angular.equals(issue, issueToResolve);
+        }) : -1;
+      if (issueIndex === -1) {
+        var invalidIssueError = new Error(
+          'An issue which was not fetched from the backend has been ' +
+          'resolved');
+        return new Promise((reject) => reject(invalidIssueError));
```

You can just throw an error, Javascript will transform it into a "reject" for you (can confirm with unit tests)

---

> 1. Go through the travis log and see what's failing.
> 2. Run that individual test suite locally on your machine to see if it failing there too.
> 3. Then let's discuss what's the issue why it's failing. I'll help you here..

Okay @bansalnitish  will try the above mentioned points.